three-cards-divination/src/main/java/com/cyber/mysticism/tarot/service/ThreeCardsDivinationService.java

```diff
@@ -87,4 +117,27 @@ private List<Card> extractCardsWithFilter(String duplicatedCardName) {
         Stream<Card> cards = getCards().stream();
         return cards.filter(card -> !duplicatedCardName.equals(card.name())).collect(toShuffledStream()).limit(2).toList();
     }
+
+    /**
+     * @return a map containing a unique reading for a particular user
+     */
+    private Map<String, Card> getUniqueReadingBasedOnHashCode(TarotUser retrievedUser, Map<String, Card> threeCardsDivinationReading) throws DivinationException {
+        String hashCode = getHashCode(threeCardsDivinationReading);
+        String[] hashCodeArray = {hashCode};
+        while (retrievedUser.getReadings().stream().anyMatch(x -> x.getHashCode().equals(hashCodeArray[0]))) {
```

Comment:
make a query to find the hash_code in db, do not iterate in code over all the results